community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/schema/NativeLuceneFusionSchemaIndexProviderFactory.java

```diff
@@ -87,7 +88,9 @@ public static FusionSchemaIndexProvider newInstance( PageCache pageCache, File s
     {
         IndexDirectoryStructure.Factory childDirectoryStructure = subProviderDirectoryStructure( storeDir );
         boolean readOnly = isReadOnly( config, operationalMode );
-        NumberSchemaIndexProvider nativeProvider =
+        StringSchemaIndexProvider stringProvider =
```

Comment:
should we introduce a feature flag that will gonna allow us to switch native sting indexes and fallback to lucene?